### PR TITLE
Temple Interior and Exterior

### PR DESCRIPTION
**Pull request by bwrsandman**
Extracting commits from https://github.com/openblack/openblack/pull/489 and clean-up.

image

image

Depends on:
- [x] https://github.com/openblack/openblack/pull/559
- [x] #634

### CODE REVIEW

**Comment by bwrsandman** [comment]
In vulkan, when I enter the temple, I get an error about the scratch buffer limit being exceeded.

File: test/mock/CMakeLists.txt [hunk 1]
```diff
@@ -78,13 +91,15 @@ add_custom_target(
   COMMAND ${CMAKE_COMMAND} -E make_directory ${TEXTURES_BINARY_DIR}
   COMMAND ${CMAKE_COMMAND} -E make_directory ${CITADEL_BINARY_DIR}
   COMMAND ${CMAKE_COMMAND} -E make_directory ${CITADEL_OUTSIDE_BINARY_DIR}
+  COMMAND ${CMAKE_COMMAND} -E make_directory ${CITADEL_ENGINE_BINARY_DIR}
```
Comment (bwrsandman) [inline test/mock/CMakeLists.txt:94]: In order to work in the run tests, there must be at least one file in here

File: src/ECS/Archetypes/CitadelArchetype.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+entt::entity CitadelArchetype::Create(const glm::vec3& position, [[maybe_unused]] const std::string& affiliation,
+                                      const glm::mat4& rotation, const glm::vec3& size)
+{
+	auto& registry = Game::Instance()->GetEntityRegistry();
+	const auto entity = registry.Create();
+	registry.Assign<Transform>(entity, position, rotation, size);
+	const auto meshId = entt::hashed_string("temple/b_first_temple_l3d");
+	registry.Assign<Mesh>(entity, meshId, static_cast<int8_t>(0), static_cast<int8_t>(0));
+	return entity;
```
Comment (bwrsandman) [inline src/ECS/Archetypes/CitadelArchetype.cpp:33]: Could you also add a citadel component for the outside entity.

File: src/ECS/Archetypes/CitadelArchetype.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+#include "ECS/Components/Transform.h"
+#include "ECS/Registry.h"
+#include "Enums.h"
+#include "Game.h"
+
+using namespace openblack;
+using namespace openblack::ecs::archetypes;
+using namespace openblack::ecs::components;
+
+entt::entity CitadelArchetype::Create(const glm::vec3& position, [[maybe_unused]] const std::string& affiliation,
```
[the remark this answers is not in the document]
Comment (raffclar) [reply src/ECS/Archetypes/CitadelArchetype.cpp]: There is. I'm using it now

File: src/ECS/Archetypes/CitadelArchetype.cpp (hunk 2 above)
Comment (raffclar) [reply src/ECS/Archetypes/CitadelArchetype.cpp:32]: Added

File: test/mock/CMakeLists.txt (hunk 1 above)
Comment (raffclar) [reply test/mock/CMakeLists.txt]: What should I add here? A python script gen a zzz file?

File: src/ECS/Systems/PlayerSystemInterface.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * Copyright (c) 2018-2023 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ ******************************************************************************/
+
+#include "Enums.h"
+#include "entt/fwd.hpp"
```
Comment (bwrsandman) [inline src/ECS/Systems/PlayerSystemInterface.h:11]:
```suggestion
#include <entt/fwd.hpp>
```

**Pull request merged**